packages/lit-dev-content/src/util/gist-conversion.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {GistFiles} from '../github/github-gists.js';
+import type {SampleFile} from 'playground-elements/shared/worker-api.js';
+
+/**
+ * Additional playground metadata that we can't encode in basic gist file
+ * primitives.
+ */
+type GistPlaygroundMetadata = {
+  files: {
+    [filename: string]: {
+      /** File display order . Gist files are always returned ~alphabetically. */
+      position: number;
+      /** Whether the file is hidden. */
+      hidden?: boolean;
+    };
+  };
+};
+
+/**
+ * Filename for gist playground metadata.
+ */
+const METADATA_FILENAME = '__lit-dev-playground-metadata__.json';
+
+/**
+ * Convert an array of Playground files to a GitHub gist.
+ */
+export const playgroundToGist = (playgroundFiles: SampleFile[]): GistFiles => {
+  const metadata: GistPlaygroundMetadata = {
+    files: Object.fromEntries(
+      playgroundFiles.map(({name, hidden}, position) => {
+        const entry: GistPlaygroundMetadata['files'][string] = {position};
+        if (hidden) {
+          entry.hidden = true;
+        }
+        return [name, entry];
+      })
+    ),
+  };
+  const gistFiles: GistFiles = Object.fromEntries(
+    playgroundFiles.map(({name, content}) => [name, {content}])
+  );
+  gistFiles[METADATA_FILENAME] = {
+    content: JSON.stringify(metadata),
+  };
+  return gistFiles;
+};
+
+/**
+ * Convert a GitHub gist to an array of Playground files.
+ */
+export const gistToPlayground = (gistFiles: GistFiles): SampleFile[] => {
+  const metadataFile = gistFiles[METADATA_FILENAME];
+  let metadata: GistPlaygroundMetadata = {files: {}};
+  if (metadataFile) {
+    try {
+      metadata = JSON.parse(metadataFile.content) as GistPlaygroundMetadata;
```

Comment:
I don't know if this is actionable, but we don't validate the structure of this metadata file. So if someone has hand written valid json that parses it will be cast and result in runtime errors.

But I think that's also probably fine.
This came to mind when I was wondering if the `.position` would always be a number.